### PR TITLE
[hotfix][runtime] Remove private methods

### PR DESCRIPTION
**Pull request by snuyanzin**
## What is the purpose of the change

The PR removes a bunch of unused private methods
_org.apache.flink.runtime.operators.hash.CompactingHashTable#insertBucketEntryFromSearch_ which became unused
after https://github.com/apache/flink/commit/925ac1f76bb84986764495407049a77552169d84

~~_org.apache.flink.runtime.state.VoidNamespace#readResolve_
was added here https://github.com/apache/flink/commit/3b97128f05bacfb80afe4a2a49741c31ff306cd2 as unused~~

_org.apache.flink.runtime.dispatcher.runner.DefaultDispatcherRunnerTest#createDispatcherGateway_ that became unused after https://github.com/apache/flink/commit/401f5f3970c6e969f3749ee72a4c57cb3f905af2

_org.apache.flink.runtime.jobmaster.slotpool.DeclarativeSlotPoolServiceTest#matchesWithSlotContext_ which became unused after https://github.com/apache/flink/commit/b67f0a70662819e064176a418ce2b892f3fb61b4

_org.apache.flink.runtime.messages.WebMonitorMessagesTest#randomIds_ which became unused
after https://github.com/apache/flink/commit/67aad88ee025ce02053ab560f2504762f53b87d9

_org.apache.flink.runtime.scheduler.adaptive.FinishedTest.MockFinishedContext#assertNoStateTransition_
becoming unused after https://github.com/apache/flink/commit/85879bfb4e6be34481aa94bd0c215db234283ccd

_org.apache.flink.runtime.scheduler.adaptive.StopWithSavepointTest#createStopWithSavepoint_
which became after https://github.com/apache/flink/commit/d13cb056912d9011df96671c3bd60299a59a1117

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.
## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: (yes)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: ( no)
  - The S3 file system connector: ( no )

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by snuyanzin** [comment]
@flinkbot run azure

**Comment by snuyanzin** [comment]
@flinkbot run azure

**Comment by snuyanzin** [comment]
@flinkbot run azure

**Comment by snuyanzin** [comment]
@flinkbot run azure

File: flink-runtime/src/main/java/org/apache/flink/runtime/state/VoidNamespace.java [hunk 1]
```diff
@@ -56,13 +54,4 @@ public boolean equals(Object obj) {
     public String toString() {
         return getClass().getSimpleName();
     }
-
-    // ------------------------------------------------------------------------
-    //  Singleton serialization
-    // ------------------------------------------------------------------------
-
-    // make sure that we preserve the singleton properly on serialization
-    private Object readResolve() throws ObjectStreamException {
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/state/VoidNamespace.java:65]: This is wrong. https://docs.oracle.com/javase/7/docs/platform/serialization/spec/input.html#5903

Comment (snuyanzin) [reply flink-runtime/src/main/java/org/apache/flink/runtime/state/VoidNamespace.java]: Thanks for pointing that out.
my bad, I reverted it

**Pull request merged**